### DISCUSSION TITLE
Propоsal: Allow simultaneous use of this (...) and base (...) when creating a class constructor

### DISCUSSION THREAD

**Discussion by DeveloperLookBook** [dotnet/csharplang#1798] [Language Ideas] [answered]
Example:

```
class A {
   private string PropA {get; set;}
   public A() {}
   public A(string propA) this.PropA = propA;
}

class B : A {
   private string PropB_1 {get; set;}
   private string PropB_2 {get; set;}

   public B(string propB_1) => this.PropB_1 = propB_1;
   public B(string propB_1, string propB_2, string propA) : this(propB_1), base(A) // simultaneous use of this (...) and base (...) 
   {
       this.PropB_2 = propB_2;
   }
}
```

**Comment by svick** [comment]
I don't think I like that adding `base(propA)` effectively changes what the preceding `this(propB_1)` means.

Also, I'm not sure there's a good way to represent this in IL: when `B(string, string, string)` calls `B(string)`, it doesn't get a choice to tell it to skip calling `A()`.

**Comment by HaloFour** [comment, marked as answer]
A constructor *must* either call a base constructor *or* another constructor on the current class.  It cannot call both as the base constructor must only be called once.

You can very easily solve your problem above with some minor refactoring.  More complicated examples can be solved through helper methods.
